etl/src/main/java/org/pih/hivmigration/etl/sql/ProgramMigrator.groovy

```diff
@@ -0,0 +1,252 @@
+            ;
+        ''')
+
+        executeMysql("Ensure health_center is set and drop superfluous values for starting_health_center", '''
+            UPDATE hivmigration_programs_raw
+            SET
+                health_center = COALESCE(health_center, starting_health_center, 1),
+                starting_health_center = IF (
+                        starting_health_center = health_center OR
+                        health_center_transfer_date IS NULL OR
+                        health_center_transfer_date > outcome_date OR
+                        health_center_transfer_date < enrollment_date,
+                    NULL,
+                    starting_health_center)
+            ;
+        ''')
+
+        executeMysql("Load single enrollments to hivmigration_programs table", '''
+            INSERT INTO hivmigration_programs
+                (source_patient_id, location_id, enrollment_date, art_start_date, outcome_date, outcome)
+            SELECT source_patient_id, health_center, enrollment_date, art_start_date, outcome_date, outcome FROM hivmigration_programs_raw
+            WHERE starting_health_center IS NULL
+            ;
+        ''')
+
+        executeMysql("Load initial enrollments into hivmigration_programs table", '''
+            insert into hivmigration_programs
+                (source_patient_id, location_id, enrollment_date, art_start_date, outcome_date, outcome)
+            SELECT
+                source_patient_id,
+                starting_health_center,
+                enrollment_date,
+                IF (art_start_date < health_center_transfer_date, art_start_date, NULL),
+                health_center_transfer_date,
+                'TRANSFERRED_INTERNALLY'
+            FROM hivmigration_programs_raw
+            WHERE starting_health_center IS NOT NULL
+            ;
+        ''')
+
+        executeMysql("Load current enrollments into hivmigration_programs table", '''
+            insert into hivmigration_programs
+                (source_patient_id, location_id, enrollment_date, art_start_date, outcome_date, outcome)
+            SELECT
+                source_patient_id,
+                health_center,
+                health_center_transfer_date,
+                IF (art_start_date < health_center_transfer_date, health_center_transfer_date, art_start_date),
+                outcome_date,
+                outcome
+            FROM hivmigration_programs_raw
+            WHERE starting_health_center IS NOT NULL
+            ;
+        ''')
+
+        executeMysql("Load to patient_program table", '''
+            insert into patient_program
+                (patient_program_id, patient_id, program_id, date_enrolled, date_completed, location_id, outcome_concept_id, creator, date_created, uuid)
+            select
+                h.patient_program_id, 
+                p.person_id,
+                2,
```

Comment:
I'd think we would want to actually look this up by uuid, not assume the primary key will be 2 for the "HIV Program"

Reply:
Oh yeah, indeed, it was enrolling everyone in the Asthma program...